### PR TITLE
Host program support

### PR DESCRIPTION
**Pull request by mikem8361**
Added the --host-only option that downloads the host program usually "dotnet"
or "apphost" renamed to the program/project name for self-contained apps.

### CODE REVIEW

**Comment by mikem8361** [comment]
Ping, just in case you miss this.

File: src/Microsoft.FileFormats/PE/PEStructures.cs [hunk 1]
```diff
@@ -99,6 +99,19 @@ public enum ImageFileMachine
         Arm64       = 0xAA64
     }
 
+    /// <summary>
+    /// Characteristics (IMAGE_FILE)
+    /// </summary>
+    [Flags]
+    public enum ImageFile : ushort
+    {
+        RELOCS_STRIPPED         = 0x0001,
```
Comment (noahfalk) [inline src/Microsoft.FileFormats/PE/PEStructures.cs]: Nit: We've typically used Pascal casing following BCL coding style rules

File: src/Microsoft.SymbolStore/KeyGenerators/KeyGenerator.cs [hunk 2]
```diff
@@ -39,6 +41,11 @@ public enum KeyTypeFlags
         /// the Portable and Windows PDBs are available on the symbol server.
         /// </summary>
         ForceWindowsPdbs = 0x08,
+
+        /// <summary>
+        /// Generate keys for the host program.
```
Comment (noahfalk) [inline src/Microsoft.SymbolStore/KeyGenerators/KeyGenerator.cs]: Can we add a little more description about what the 'host program' relationship is? I assume for a binary 'foo', it is a key with the same buildid renamed to 'apphost'?

File: src/Microsoft.FileFormats/PE/PEStructures.cs (hunk 1 above)
Comment (mikem8361) [reply src/Microsoft.FileFormats/PE/PEStructures.cs]: I was being lazy and copied this enum from clrmd. I'll fix it.

File: src/Microsoft.SymbolStore/KeyGenerators/KeyGenerator.cs (hunk 2 above)
Comment (mikem8361) [reply src/Microsoft.SymbolStore/KeyGenerators/KeyGenerator.cs]: Done.

**Pull request merged**